test/endpoint_provisioner_test.go

```diff
@@ -0,0 +1,1083 @@
+
+var _ = Describe("EndpointProvisioner", func() {
+	var messagingService solace.MessagingService
+	BeforeEach(func() {
+		var err error
+		messagingService, err = messaging.NewMessagingServiceBuilder().
+			FromConfigurationProvider(helpers.DefaultConfiguration()).
+			WithProvisionTimeoutMs(1 * time.Millisecond).
+			Build()
+		Expect(err).To(BeNil())
+	})
+
+	It("fails to provision when given a max message redelivery range < 0", func() {
+		outcome := messagingService.EndpointProvisioner().FromConfigurationProvider(config.EndpointPropertyMap{
+			config.EndpointPropertyMaxMessageRedelivery: -1,
+		}).Provision(provisionQueueName, true)
+		Expect(outcome.GetError()).To(HaveOccurred())
+		Expect(outcome.GetError()).To(BeAssignableToTypeOf(&solace.IllegalStateError{}))
+		Expect(outcome.GetStatus()).To(BeFalse())
+	})
+
+	It("fails to provision when given a max message redelivery range > 255", func() {
+		outcome := messagingService.EndpointProvisioner().FromConfigurationProvider(config.EndpointPropertyMap{
+			config.EndpointPropertyMaxMessageRedelivery: 256,
+		}).Provision(provisionQueueName, true)
+		Expect(outcome.GetError()).To(HaveOccurred())
+		Expect(outcome.GetError()).To(BeAssignableToTypeOf(&solace.IllegalStateError{}))
+		Expect(outcome.GetStatus()).To(BeFalse())
+	})
+
+	It("fails to provision when given an invalid max message size", func() {
+		outcome := messagingService.EndpointProvisioner().FromConfigurationProvider(config.EndpointPropertyMap{
+			config.EndpointPropertyMaxMessageSize: -1,
+		}).Provision(provisionQueueName, true)
+		Expect(outcome.GetError()).To(HaveOccurred())
+		Expect(outcome.GetError()).To(BeAssignableToTypeOf(&solace.IllegalStateError{}))
+		Expect(outcome.GetStatus()).To(BeFalse())
+	})
+
+	It("fails to provision when given an invalid queue quota size", func() {
+		outcome := messagingService.EndpointProvisioner().FromConfigurationProvider(config.EndpointPropertyMap{
+			config.EndpointPropertyQuotaMB: -1,
+		}).Provision(provisionQueueName, true)
+		Expect(outcome.GetError()).To(HaveOccurred())
+		Expect(outcome.GetError()).To(BeAssignableToTypeOf(&solace.IllegalStateError{}))
+		Expect(outcome.GetStatus()).To(BeFalse())
+	})
+
+	It("fails to provision when given an invalid queue permission", func() {
+		outcome := messagingService.EndpointProvisioner().FromConfigurationProvider(config.EndpointPropertyMap{
+			config.EndpointPropertyPermission: "not a permission",
+		}).Provision(provisionQueueName, true)
+		Expect(outcome.GetError()).To(HaveOccurred())
+		Expect(outcome.GetError()).To(BeAssignableToTypeOf(&solace.IllegalStateError{}))
+		Expect(outcome.GetStatus()).To(BeFalse())
+	})
+
+	It("fails to provision when given valid queue properties with unstarted messaging service", func() {
+		outcome := messagingService.EndpointProvisioner().FromConfigurationProvider(config.EndpointPropertyMap{
+			config.EndpointPropertyDurable:   false, // this will be overrided to true
```

Comment:
Does the user know that this is happening if they set it? In other APIs (JS comes to mind from recent dev) the call to provision is rejected if durability is set to false

Reply:
@oodigie this for sure needs to change, from the arch "Provisioning temporary endpoints through this new interface (without queue template trickery) will initially be disabled..."

Reply:
From the ARCH (and ARCH review meetings) we decided to expose the `WithDurability()` method and its corresponding property in the properties map but default it to True. I am not sure if this has changed.

Reply:
Not sure that this is what we discussed in our ARCH review. We agreed to set it to True even when the false is passed to the method. We should agree which decision we want to take in this scenario.

Reply:
The test plan specifically calls our trying this with false and expecting an exception/error.  In JS we raised https://sol-jira.atlassian.net/browse/SOL-116163 because the error was incorrect but we still expect something

Reply:
@sdewilde, I will update the Go API to match returning an error like in SolclientJS. I will have to update the ARCH to state this.
cc: @cjwmorgan-sol

Reply:
Done